src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
+
+  men.forEach((m) => sum += m.died - m.born);
+
```

Comment:
We can even create a reusable function to calculate men and women sum result like:

`const calculateAverageAge = (filteredPeople) => {`
  `const sum = filteredPeople.reduce((total, p) => total + (p.died - p.born), 0);`
  `return sum / filteredPeople.length;`
`}`

Then using be: `return calculateAverageAge(filteredMen)` instead of `return sum/men.length`